content/post/ddos-detection.md

```diff
@@ -0,0 +1,413 @@
++++
+title = "DDoS Attack Detection with Wallaroo: A Real-time Time Series Analysis Example"
+draft = false
+date = 2017-11-24T00:00:00Z
+tags = [
+    "wallaroo",
+    "windowing",
+    "partitioning",
+    "python",
+    "example",
+    "tutorial",
+    "ddos",
+    "timeseries",
+    "anomaly detection"
+]
+categories = [
+    "Windowing",
+    "Examples"
+]
+description = "Building a DDoS attack detector using timeseries analysis in real time with Wallaroo"
+author = "nisanharamati"
++++
+
+
+# DDoS Attack Detection with Wallaroo: A Real-time Time Series Analysis Example
+
+In this post, I would like to demonstrate a real-world use case for Wallaroo: DDoS attack detection in real time, using a stream of request logs from a large group of servers. This example builds on the concepts introduced in [Non-native event-driven windowing in Wallaroo][windowing in wallaroo], adding partitioning, parallelization, and the application of statistical analysis to perform a simplified anomaly detection based on weighted mean and standard deviation to determine when a server is experiencing an uncharacteristic and heavy load.
+
+If you're new to Wallaroo and are interested in learning more about it before diving into today's example, you can check our [Go Python, Go! Stream Processing for Python][go python] post, where we introduce the Wallaroo Python API.
+
+The code for the example in this post can be found on [GitHub][demo code], and can be run with the [current version of Wallaroo (0.2.2)](https://github.com/WallarooLabs/wallaroo/tree/0.2.2).
+
+## Your Service Layer is Under Attack
+
+Running a large application over the internet is really great! You can reach a massive customer base, and provide them with your digital product in no time at all. Unless...
+
+![503][503 service unavailable]
+(source: https://andrewodendaal.com/503-service-unavailable-back-end-server-is-at-capacity/)
+![502 bad gateway][502 bad gateway]
+![this site cannot be reached][this site cannot be reached]
+
+Well that sucks.
+
+There are a lot of reasons why a service can become unavailable or unreachable, and many of them are benign, like hardware hiccups, or bad code, or maybe a messed up routing table; one reason, however, is not benign: a denial of service (DoS) attack, or more commonly these days, a distributed denial of service (DDoS) attack.
```

Comment:
Given this is about DDoS and my time is important, I think you should get right to it. You just took up a lot of space on the page to not address the topic at hand. I think you can lose a bunch of people by here.

There's a long intro paragraph where its easy for me to miss that this is about DDoS. For a product I've never heard of and you don't explain to me and then all this. 

I think you've lost a lot of people by now. The lead in to "lets talk about DDoS and Wallaroo" needs to be more information dense and needs to move quicker.